src/defrag.c

```diff
@@ -381,26 +401,14 @@ void activeDefragSdsDictCallback(void *privdata, const dictEntry *de) {
 
 void activeDefragHfieldDictCallback(void *privdata, const dictEntry *de) {
     dict *d = privdata;
-    hfield newhf, hf = dictGetKey(de);
+    hfield hf = dictGetKey(de);
 
     if (hfieldGetExpireTime(hf) == EB_EXPIRE_TIME_INVALID) {
         /* If the hfield does not have TTL, we directly defrag it. */
-        newhf = activeDefragHfield(hf);
+        activeDefragHfield(hf, d);
```

Comment:
Cannot we use here `de` to save the lookup for hfield?

Reply:
good idea.

Reply:
done with https://github.com/redis/redis/pull/13842/commits/175115f1c13fe6348ac736a0f5a4a41ade2c0c24